PULL_REQUEST_TEMPLATE.md

```diff
@@ -6,4 +6,4 @@ Please refer to our [guidelines for making contributions](https://github.com/jbo
 
 The build axis can be controlled by prefixing a ! on the following as appropriate:
 
-MAIN CORE TOMCAT AS_TESTS RTS JACOCO XTS QA_JTA QA_JTS_JACORB QA_JTS_JDKORB QA_JTS_OPENJDKORB PERF LRA !NO_WIN DB_TESTS mysql db2 postgres oracle
+MAIN CORE TOMCAT AS_TESTS RTS JACOCO XTS QA_JTA QA_JTS_JACORB QA_JTS_JDKORB QA_JTS_OPENJDKORB PERF LRA_THORNTAIL LRA_WILDFLY !NO_WIN DB_TESTS mysql db2 postgres oracle
```

Comment:
Under which circumstances would we make a change to LRA that would *not necessitate* running on all supported platforms. It seems niche to force a submitter to have to understand what all these axes mean. I think anyone, apart from the team, doesn't touch these because they are all niche. Maybe we need one or two variables which are clear and which under the covers starts a group of axes, and for those in the know such as ourselves we can slip in other variables. For example:

!LRA implies don't run any LRA axes and if you say make a wildfly specific change you would use LRA_WILDFLY etc

Reply:
+1

I agree that a single profile is easier to understand for the community and I also think we should test always on both platforms (i.e Wildfly and Thorntail/Quarkus). I am not sure about adding extra variables behind the scene though: I would be fine to start both profiles when LRA is tested (it does not take much time to test both Wildfly and Quarkus)

Reply:
I'll repeat my motivation; we want to aim to limit the number of options available to the person raising the PR. So the following would satisfy that goal:

LRA enables both axes. Then if one of the team wants to skip thorntail they would add !LRA_THORNTAIL. Similarly for !LRA_WILDFLY. But I'm a little unsure here as well, isn't LRA_THORNTAIL for running the TCK and LRA_WILDFLY for running integration tests and if so we should use LRA_TCK and LRA_INTEGRATION, in this way we abstract away the details of our (private) CI infrastructure and we also won't need to change the scripts when we remove thorntail.

The key point being that the person raising the PR is generally only interested in asking for LRA to be tested not the internals of what gets tested. We have too many axes being exposed to people raising PRs.

Reply:
LRA TCK gets tested with WFLY and QUARKUS. The two profiles LRA_THORNTAIL (maybe this should be renamed to LRA_QUARKUS) and LRA_WILDFLY just select where `lra-coordinator` will be deployed (i.e. QUARKUS or WILDFLY respectively)

IMO, testing a PR on both WFLY and QUARKUS should be mandatory as something that works on one of them could not work on the other (which is the actual scenario we have in the main branch)

Reply:
Ok.

Reply:
> LRA TCK gets tested with WFLY and QUARKUS. The two profiles LRA_THORNTAIL (maybe this should be renamed to LRA_QUARKUS) and LRA_WILDFLY just select where `lra-coordinator` will be deployed (i.e. QUARKUS or WILDFLY respectively)
> 
> IMO, testing a PR on both WFLY and QUARKUS should be mandatory as something that works on one of them could not work on the other (which is the actual scenario we have in the main branch)

Ondra said he found an issue when running the TCK with quarkus?

Reply:
> If we aren't using thorntail can we get rid of all of the references to it?

We are still using Thorntail as an Arquillian container to host `lra-participant`. I haven't checked your PR in `jbosstm/quickstart` but if Quarkus has developed a valid Arquillian container then the answer is yes :-)

Reply:
I have no idea but I will mark it as approved once you have gone back to just one axis for LRA testing.

Reply:
I am marking this as resolved since Manuel's changes have addressed both of our (mine and Ondra) comments